### DISCUSSION TITLE
Gif images always load from network

### DISCUSSION THREAD

**Discussion by qwaskhaled** [coil-kt/coil#872] [Q&A]
I am using gif decoders to support gif loading. However, I am having an issue that the gif images always keep loading from the network instead of memory or disk after the first time they get fetched from the server

I expect the images loader to load the gif image from the network one time and then fetch it from disk or cache when it's needed again. 

To replicate the issue create a recycler view that contains gif images and start scrolling it. You will notice the same images will gets loaded again from the network.

I am using Android 11 and the library version 1.3.2. It's worth mentioning that I have enabled the disk caching policy and it works fine for all images except for git. 

Is there anyway I can reslove that?

**Comment by colinrtwhite** [comment]
It's likely the cache control header for the image isn't allowing it to be cached in the disk cache. If you control the server, you can ask them to change the `cache-control` header. Else, you can overwrite it locally using an OkHttp interceptor (you can copy [the code from the sample app](https://github.com/coil-kt/coil/blob/1.x/coil-sample/src/main/java/coil/sample/Application.kt#L51) which does this).

**Reply by qwaskhaled** [reply]
No, it's not working. Is there any way to do manual caching for gif images? I did it for regular images with a help of LRU cache and interceptors but I can't do for gif they get static images.

**Reply by colinrtwhite** [reply]
What do you mean by "manual caching"? Also GIFs shouldn't be static; they're rendered using either a `MovieDrawable` or `AnimatedImageDrawable` depending on the API level.

**Comment by 4z7l** [comment]
I have similar issue. How did you solve it? (Cache-control doesn't work either)
